### PR TITLE
Cairo contract to mint aBTC

### LINKED ISSUE #34: Cairo contract to mint aBTC
Cairo contract to mint aBTC with a one-to-one mint with wBTC.

Create the mintVaultBTC and one contract for a reserve of wBTC in one contract. 

- [] Architecture
- [] ERC20 Mintable with AccessControl from OpenZeppelin
- [] Deposit function
- [] Withdraw function

Bonus: Upgradeable contract

### PR DESCRIPTION
**Pull request by manlikeHB**
closes #34

### CODE REVIEW

File: onchain/cairo/src/tokens/erc20_mintable.cairo [hunk 1]
```diff
@@ -83,8 +83,13 @@ mod ERC20Mintable {
     #[abi(embed_v0)]
     impl IERC20MintableImpl of super::IERC20Mintable<ContractState> {
         fn mint(ref self: ContractState, recipient: ContractAddress, amount: u256) {
-            self.ownable.assert_only_owner();
+            // self.ownable.assert_only_owner();
```
Comment (manlikeHB) [inline onchain/cairo/src/tokens/erc20_mintable.cairo:86]: @MSghais, owner should be set to vault contract address right?

File: onchain/cairo/src/tests/vault_tests.cairo [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+    fn SYMBOL(symbol: felt252) -> felt252 {
+        'symbol'.try_into().unwrap()
+    }
+
+    fn setup() -> (IERCVaultDispatcher, IERC20Dispatcher, IERC20Dispatcher,) {
+        let erc20mintable_class = declare("ERC20Mintable").unwrap();
+        let erc20_class = declare("ERC20").unwrap();
+
+        let wBTC_Dispathcer = deploy_erc20(
```
Comment (MSghais) [inline onchain/cairo/src/tests/vault_tests.cairo:36]: Can you fix typo and not use camelCase please?

File: onchain/cairo/src/tokens/erc20_mintable.cairo (hunk 1 above)
Comment (MSghais) [reply onchain/cairo/src/tokens/erc20_mintable.cairo:86]: Yes let's add an Access Control init in the constructor to the ERC20 Mintable, and also a set_role to add others role grant access, check example here:

// Admin
// Add OPERATOR role to the Vault escrow
// #[external(v0)]
// fn set_control_role(
//     ref self: ContractState, recipient: ContractAddress, role: felt252, is_enable: bool
// ) {
//     self.accesscontrol.assert_only_role(ADMIN_ROLE);
//     assert!(
//         role == ADMIN_ROLE
//             || role == OPERATOR_ROLE // Think and Add others roles needed on the protocol
//             ,
//         "role not enable"
//     );
//     if is_enable {
//         self.accesscontrol._grant_role(role, recipient);
//     } else {
//         self.accesscontrol._revoke_role(role, recipient);
//     }
// }

File: onchain/cairo/src/defi/vault.cairo [hunk 3]
```diff
@@ -101,23 +135,67 @@ mod Vault {
...
+
+            // Receive/burn token minted
+            let token_mintable = IERC20MintableDispatcher {
+                contract_address: self.token_address.read()
+            };
+            token_mintable.burn(caller, amount);
+
+            // Resend amount of coin deposit by user
+            let token_deposited = IERC20Dispatcher { contract_address: token_address };
+            let amount_ratio = amount / self.token_permitted.read(token_address).ratio_mint;
```
Comment (MSghais) [inline onchain/cairo/src/defi/vault.cairo:149]: Please comment with a TODO, and just add the amount without the ratio at the moment

File: onchain/cairo/src/defi/vault.cairo [hunk 4]
```diff
@@ -76,22 +76,56 @@ mod Vault {
...
+            token_deposited.transfer_from(caller, get_contract_address(), amount);
+
+            // Mint token and send it to the receiver
+            let token_mintable = IERC20MintableDispatcher {
+                contract_address: self.token_address.read()
+            };
+            let token = self.token_permitted.read(token_address);
+
+            // Calculate the ratio if 1:1, less or more
+            let amount_ratio = token.ratio_mint * amount;
```
Comment (MSghais) [inline onchain/cairo/src/defi/vault.cairo:95]: Please comment with a TODO, and just add the amount without the ratio at the moment

**Review by MSghais — CHANGES_REQUESTED** [review]

**Review by MSghais — APPROVED** [review]

**Pull request merged**